runtime/appruntime/apisdk/api/gateway.go

```diff
@@ -36,18 +38,58 @@ func (s *Server) createGatewayHandlerAdapter(h Handler) httprouter.Handle {
 	// but locally we don't want the overhead of logging every request.
 	logger := s.rootLogger.With().Str("service", service.Name).Str("endpoint", h.EndpointName()).Logger()
 
-	proxy := httputil.NewSingleHostReverseProxy(serviceBaseURL)
-	proxy.ErrorLog = newZeroLogAdapter(logger, zerolog.ErrorLevel)
-	proxy.ErrorHandler = func(w http.ResponseWriter, req *http.Request, err error) {
-		logger.Err(err).Msg("error proxying request to service")
-		errs.HTTPError(w, errs.B().Cause(err).Code(errs.Unavailable).Err())
-	}
-
+	proxy := s.createProxyToService(service, h.EndpointName(), serviceBaseURL, logger)
 	return func(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-		if s.runtime.EnvCloud != "local" {
-			logger.Trace().Msg("proxying request to service")
+		s.beginOperation()
+		defer s.finishOperation()
+
+		meta := CallMetaFromContext(req.Context())
+
+		info, proceed := s.runAuthHandler(h, s.NewIncomingContext(w, req, toUnnamedParams(ps), meta))
+		if proceed {
+			meta.Internal = &InternalCallMeta{
+				Caller: GatewayCaller{
+					ServiceName: h.ServiceName(),
+					Endpoint:    h.EndpointName(),
+				},
+				AuthUID:  string(info.UID),
```

Comment:
Are we guaranteed that `info` is non-nil if `proceed` is true? (I forget the semantics of `runAuthHandler`)

Reply:
`info` isn't a pointer - thus we always garanutee at least a zero value info struct here.